platform/mds/mds/src/main/java/org/motechproject/mds/annotations/Discriminated.java

```diff
@@ -0,0 +1,13 @@
+package org.motechproject.mds.annotations;
+
+import javax.jdo.annotations.IdentityType;
+import javax.jdo.annotations.PersistenceCapable;
+import java.lang.annotation.*;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@PersistenceCapable(identityType = IdentityType.DATASTORE, detachable = "true")
+public @interface Discriminated {
```

Comment:
Deciding what relationship MOTECH-1768 has to inheritance is a good question.  Here are a couple of points on MDS @Discriminator vs JDO @Inheritance:

1) My experience in metadata design is that verbatim inclusion of foreign metadata items (JDO @Inheritance) within a set of more tightly curated items (@Entity, @Field, all other MDS things) has a tendency to create pathologies about what combinations mean.  Often the pathologies are hard to identify up front, or only come into play when future changes are made.

2) Here is a dividing line I propose:

- Using @Discriminated means you can't also use @Inheritance on the same entity.
- If desired, sharing fields can be achived by java interface/implements, along with @Discriminated.  This presumes that the datanucleus enhancer can pass over the interfaces without getting confused.  Enhancer passing over seems plausible to me, but the datanucelus sources should probably be checked.